services/src/main/java/org/keycloak/services/resources/WelcomeResource.java

```diff
@@ -302,4 +310,14 @@ private void csrfCheck(final MultivaluedMap<String, String> formData) {
         }
     }
 
+    protected boolean createsTemporaryAdminUser() {
+        try {
+            String themeProperty = "kcBootstrapAdminIsTemporary";
```

Comment:
The method name should be createAdminUser, right?

Reply:
I renamed it to `isAdminUserTemporary` to make it more descriptive. In general, this returns `true` if the Welcome Page should create a temporary user.